lib/src/evaluation/evaluator.dart

```diff
@@ -16,77 +16,75 @@ import 'evaluator_options.dart';
 
 import 'manager_messages.dart';
 
+/// A type alias for an action to be performed with a temporary evaluator.
 typedef EvaluationAction<T> = Future<T> Function(Evaluator);
 
-/// A convenience method for running an action given an evaluator with the supplied evaluator options.
-/// After [action] completes, the evaluator is closed.
-///
-///   - [options]: The options used to configure the evaluator.
-///   - [action]: The action to perform.
-Future<T> withEvaluator<T>(EvaluatorOptions options, EvaluationAction action) async {
-  return await withEvaluatorManager((manager) async {
-    final evaluator = await manager.newEvaluator(options: options);
-    return await action(evaluator);
-  });
-}
-
-/// Like [withEvaluator] with options, but with preconfigured evaluator options.
-///
-/// - [action]: The action to perform
-Future<T> withEvaluatorPreconfigured<T>(EvaluationAction action) async {
-  return await withEvaluator(await EvaluatorOptions.preconfigured, action);
-}
-
-/// Like [withProjectEvaluator] with options, but configured with preconfigured options.
-///
-///   - [projectBaseUri]: The base path containing the PklProject file.
-///   - [action]: The action to perform.
-/// - Returns: The result of the action.
-Future<T> withProjectEvaluatorPreconfigured<T>(Uri projectBaseUri, EvaluationAction action) async {
-  return await withProjectEvaluator(projectBaseUri, await EvaluatorOptions.preconfigured, action);
-}
-
-/// Convenience method for initializing an evaluator from the project.
-///
-/// [options] is the base set of evaluator options.
-/// Any `evaluatorSettings` set within the PklProject file overwrites any fields set on [options].
-///
-/// After [action] completes, the evaluator is closed.
-/// - Parameters:
-///   - [projectBaseUri]: The base path containing the PklProject file.
-///   - [options]: The base options used to configure the evaluator.
-///   - [action]: The action to perform.
-/// - Returns: The result of the action.
-Future<dynamic> withProjectEvaluator<T>(
-  Uri projectBaseUri,
-  EvaluatorOptions options,
-  EvaluationAction action,
-) async {
-  return await withEvaluatorManager((manager) async {
-    final evaluator = await manager.newProjectEvaluator(
-      projectBaseUri: projectBaseUri,
-      options: options,
-    );
-    final result = await action(evaluator);
-    await evaluator.close();
-    return result;
-  });
-}
-
 /// The core API for evaluating Pkl modules.
+///
+/// Use the static `run` methods for one-off evaluations with automatic resource
+/// management. For managing multiple, long-lived evaluators, use an
+/// [EvaluatorManager] instance directly.
```

Comment:
nit: We can make these comments more descriptive, with code examples if any.
```suggestion
/// The core API used for interfacing with Pkl and evaluating Pkl expressions and modules
///
/// You can evaluate complete modules using the [evaluateModule] function, or evaluate expressions in a module using the [evaluateExpression] function.
/// You can also convert these results to Dart Types by passing a [PklDecodable] factory and using [evaluateModuleAs] and [evaluateExpressionAs] respectively.
/// ```dart
/// void main() {
///  Evaluator.run((evaluator) {
///    /* code */
///  });
/// }
/// ```
/// To make use of short-lived evaluators with single managers, you can simply run `Evaluator.run`
/// 
/// For managing multiple evaluators, use an [EvaluatorManager] instance directly
```

Reply:
The functions have this.